model/src/test/kotlin/licenses/LicenseConfigurationTest.kt

```diff
@@ -0,0 +1,144 @@
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = false
+            )
+
+            shouldThrow<java.lang.IllegalArgumentException> {
+                LicenseConfiguration(licenseCategories = listOf(cat1, cat2), licenses = listOf(lic1, lic2))
+            }
+        }
+    }
+
+    "LicenseConfiguration" should {
+        "fetch all licenses for a specific category" {
+            val cat1 = LicenseCategory("permissive", "Permissive licenses")
+            val cat2 = LicenseCategory("non permissive", "Strict licenses")
+            val lic1 = License(
+                SpdxSingleLicenseExpression.parse("ASL-1"), sortedSetOf("permissive"),
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = false
+            )
+            val lic2 = License(
+                SpdxSingleLicenseExpression.parse("ASL-2"), sortedSetOf("permissive"),
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = true
+            )
+            val lic3 = License(
+                SpdxSingleLicenseExpression.parse("GPL"), sortedSetOf("non permissive"),
+                includeInNoticeFile = false, includeSourceCodeOfferInNoticeFile = true
+            )
+            val licenseConfiguration = LicenseConfiguration(
+                licenseCategories = listOf(cat1, cat2),
+                licenses = listOf(lic1, lic2, lic3)
+            )
+
+            val permissiveLicenses = licenseConfiguration.getLicensesForCategory(cat1.name)
+
+            permissiveLicenses shouldContainExactlyInAnyOrder setOf(lic1, lic2)
+        }
+
+        "throw an exception when querying the licenses for an unknown category" {
+            val cat = LicenseCategory("oneAndOnlyCategory")
+            val lic = License(
+                SpdxSingleLicenseExpression.parse("LICENSE"), sortedSetOf(cat.name),
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = true
+            )
+            val licenseConfiguration = LicenseConfiguration(licenses = listOf(lic), licenseCategories = listOf(cat))
+
+            shouldThrow<IllegalStateException> {
+                licenseConfiguration.getLicensesForCategory("nonExistingCategory")
+            }
+        }
+
+        "allow querying a license by ID" {
+            val lic1 = License(
+                SpdxSingleLicenseExpression.parse("ASL-1"), emptySortedSet(),
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = false
+            )
+            val lic2 = License(
+                SpdxSingleLicenseExpression.parse("ASL-2"), emptySortedSet(),
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = true
+            )
+            val licenseConfiguration = LicenseConfiguration(licenses = listOf(lic1, lic2))
+
+            licenseConfiguration[SpdxSingleLicenseExpression.parse("ASL-2")] shouldBe lic2
+        }
+
+        "allow access to the names of all categories" {
```

Comment:
`categoryNames should contain the expected licenses.

Reply:
Added more sections and renamed tests.